storage/kv/badger.go

```diff
@@ -91,6 +92,11 @@ func createDB(logger *zap.Logger, options basedb.Options, inMemory bool) (*Badge
 		go badgerDB.periodicallyCollectGarbage(logger, options.GCInterval)
 	}
 
+	out := z.CallocNoRef(1, "jemalloc check")
+	defer z.Free(out)
+	jemallocEnabled := len(out) > 0
+	logger.Debug("jemalloc allocator will be used", zap.Bool("jemalloc_enabled", jemallocEnabled))
```

Comment:
This is pretty much the best way [I found](https://github.com/dgraph-io/badger/blob/7f657f862cbde20503d33a546cdfc928e8d833cd/badger/main.go#L47-L50) to check if `jemalloc` is actually used or not,

note, we print **true** both for `-tags="blst_enabled,jemalloc"` and `-tags="blst_enabled,jemalloc,allocator"`, and hence it reports on the presence of jemalloc-only (nothing about `allocator`), 

`allocator` is a "different strategy" to approach memory management (from what I understand, it serves as a kind of cache between go application and jemalloc buffering large chunks of memory - hence it will make less cgo calls but will be holding onto large unused memory chunks increasing overall memory consumption):
```go
// Allocator amortizes the cost of small allocations by allocating memory in
// bigger chunks.  Internally it uses z.Calloc to allocate memory. Once
// allocated, the memory is not moved, so it is safe to use the allocated bytes
// to unsafe cast them to Go struct pointers. Maintaining a freelist is slow.
// Instead, Allocator only allocates memory, with the idea that finally we
// would just release the entire Allocator.
type Allocator struct {
	sync.Mutex
	compIdx uint64 // Stores bufIdx in 32 MSBs and posIdx in 32 LSBs.
	buffers [][]byte
	Ref     uint64
	Tag     string
}
```
It's hard to tell without doing any tests (or me not having enough context on how exactly we use Badger), but I think if we want to reduce memory footprint as much as possible - we better go with `-tags="blst_enabled,jemalloc"`.

Reply:
> It's hard to tell without doing any tests (or me not having enough context on how exactly we use Badger), but I think if we want to reduce memory footprint as much as possible - we better go with -tags="blst_enabled,jemalloc"

Having done some testing on stage I would say there isn't a noticeable difference for the workloads we do (for operator managing ~500 validators), see https://github.com/ssvlabs/ssv/issues/1643#issuecomment-2447336686 for more details. So I've added `allocator` tag as well to match how it originally was configured back when we first started using `jemalloc`.